homeassistant/scripts/check_config.py

```diff
@@ -0,0 +1,172 @@
+
+        def printlist(obj, level=0, pref=''):
+            """Printlist."""
+            indent1 = level * ' '
+            indent0 = pref.rjust(level)
+            if isinstance(obj, list):
+                for ob0 in obj:
+                    printlist(ob0, level+2, '- ')
+            # elif isinstance(obj, dict):
+            #    for key0, val0 in obj.items():
+            #        printlist(val0, indent00 + key0 + ':' , indent1)
+            else:
+                print(indent(obj, indent0, indent1))
+
+        components = deep_convert_dict(res['components'])
+        for name, config in components:
+            if config is None:
+                continue
+            print(color(name + ':'))
+            printlist(config, 2, '')
+
+    return 0
+
+
+def check(config_path):
+    """Perform a check by mocking hass load functions."""
+    # List of components/platforms
+    components = []
+    yaml_files = {}
+    secrets = {}
+
+    original_get_component = loader.get_component
+    original_load_yaml = yaml.load_yaml
+
+    def mock_load_yaml(filename):
+        """Mock hass.util.load_yaml to save config files."""
+        yaml_files[filename] = True
+        return original_load_yaml(filename)
+
+    def mock_get_component(comp_name):
+        """Mock hass.loader.get_component to replace setup & setup_platform."""
+        def mock_setup(*kwargs):
+            """Mock setup, only record the component name & config."""
+            # pylint: disable=cell-var-from-loop
+            components.append((comp_name, kwargs[1].get(comp_name)))
+            return True
+
+        module = original_get_component(comp_name)
+        # Test if platform/component. Also: '.' in comp_name = platform
+        if hasattr(module, 'setup'):
+            module.setup = mock_setup
+        if hasattr(module, 'setup_platform'):
+            module.setup_platform = mock_setup
+
+        module.setup_platform = mock_setup
+        return module
+
+    @patch("homeassistant.util.yaml.load_yaml",
+           side_effect=mock_load_yaml)
+    @patch("homeassistant.loader.get_component",
+           side_effect=mock_get_component)
+    def load_all_the_config(*mocks):
+        """Load the configs with appropriate patching."""
+        # Remove black & white logger, will be replace by hass logger
+        for handler in logging.root.handlers[:]:
+            logging.root.removeHandler(handler)
+        # Ensure we skip pip for speed
+        bootstrap.from_config_file(config_path, skip_pip=True)
+
+        _LOGGER.info('Load complete')
+
+        for mock, func in zip(reversed(mocks), ['load_yaml', 'get_component']):
+            if len(mock.call_args_list) == 0:
+                _LOGGER.warning('Function %s never called', func)
+
+    load_all_the_config()
+
+    return {'yaml_files': yaml_files,
+            'secrets': secrets,
+            'components': components}
+
+
+def deep_convert_dict(layer):
```

Comment:
Still battling to print it in a presentable way and this was one of those attempts to improve printing. But ideally it should look like yaml.
I'll look into yaml.dump you suggested below for the `--info` command